postproengine/wake_meandering.py

```diff
@@ -0,0 +1,365 @@
+
+        plot:
+            xlabel: 'Y [m]'
+            ylabel: 'Z [m]'
+            iter: 0
+            savefile: wake_center_{iplane}.png
+
+        statistics:
+            savefile: wake_stats_{iplane}.csv
+            mean: True
+            std: True
+            anisotropy: True
+    """
+    actionlist = {}                    # Dictionary for holding sub-actions
+
+    # --- Stuff required for main task ---
+    def __init__(self, inputs, verbose=False):
+        self.yamldictlist = []
+        inputlist = inputs if isinstance(inputs, list) else [inputs]
+        for indict in inputlist:
+            self.yamldictlist.append(mergedicts(indict, self.inputdefs))
+        if verbose: print('Initialized '+self.name)
+        return
+    
+    def execute(self, verbose=False):
+        if verbose: print('Running '+self.name)
+        # Loop through and create plots
+        for entry in self.yamldictlist:
+            iplanes  = entry['iplane']
+            trange   = entry['trange']
+            ncfile   = entry['ncfile']
+            group    = entry['group']
+            self.yhub     = entry['yhub']
+            self.zhub     = entry['zhub']
+            diam     = entry['diam']
+            Ct       = entry['Ct']
+            Uinf     = entry['Uinf']
+            method   = entry['method']
+            savefile = entry['savefile']
+            varnames = ['velocityx','velocityy','velocityz']
+            self.output_dir =  entry['output_dir']
+
+            if not isinstance(iplanes, list): iplanes = [iplanes,]
+            #Get all times if not specified 
+            filelist = []
+            for fileiter in range(0,len(ncfile)):
+                filelist.append(ncfile[fileiter])
+
+            udata = {}
+            xc = {}
+            timevec = ppsample.getVar(ppsample.loadDataset(ncfile), 'time')
+            if trange==[]:
+                # load entire file
+                self.db = ppsamplexr.getFullPlaneXR(ncfile,len(timevec),timevec[1]-timevec[0],groupname=group)
+                y = np.asarray(self.db['y'].data)
+                z = np.asarray(self.db['z'].data)
+                t = np.asarray(np.array(self.db['times']).data)
+                for iplane in iplanes:
+                    xc[iplane] = np.asarray(self.db['x'].data[iplane])
+                    udata[iplane] = np.zeros((len(t),len(z),len(y),3))
+                    udata[iplane][:,:,:,0] = np.array(self.db['velocityx'][:,iplane,:,:])
+                    udata[iplane][:,:,:,1] = np.array(self.db['velocityy'][:,iplane,:,:])
+                    udata[iplane][:,:,:,2] = np.array(self.db['velocityz'][:,iplane,:,:])
+            else:
+                find_nearest = lambda a, a0: np.abs(np.array(a) - a0).argmin()
+                iters = [find_nearest(timevec, t) for t in trange]
+                iters = np.arange(iters[0],iters[1]+1)
+                self.db = ppsamplexr.getPlaneXR(ncfile,iters,varnames,groupname=group,verbose=0,includeattr=False,gettimes=True)
```

Comment:
getPlaneXR only takes a single netcdf file, not a list of files like avgPlaneXR correct? So we can only compute wake center timeseries for the time range in a single netcdf file

Reply:
Yes, that's right, `getPlaneXR()` only handles a single netcdf file at a time.  Ideally we should upgrade it to do a series of files, but we can put that on the list of things to do.

Reply:
Sounds good. Similarly, the code between lines 162-188 could be simplified at some point too.